qiskit/algorithms/eigen_solvers/numpy_eigen_solver.py

```diff
@@ -263,3 +322,64 @@ def compute_eigenvalues(
 
         logger.debug("EigensolverResult:\n%s", self._ret)
         return self._ret
+
+    def compute_transition_amplitudes(
+        self,
+        aux_operators: Optional[ListOrDict[OperatorBase]],
+        transition_amplitude_pairs: Optional[ListOrDict[Tuple[int, int]]],
+    ) -> Optional[ListOrDict[Tuple[complex, complex]]]:
+        """Computes the transition amplitudes for the desired auxiliary operators and eigenstates.
+        Args:
+            aux_operators: ListOrDict of auxiliary operators.
+            transition_amplitude_pairs: Dictionary specifying the names of the auxiliary operators
+            to consider, and the pairs of indices to take into account for the eigenstates.
+
+        Returns:
+            Dictionary of the resulting transition amplitudes. The keys of the result are
+            constructed with the following formula: aux_str + "_" + str(i) + "_" + str(j)
+
+        .. code-block::python
+
+            from qiskit.opflow import PauliSumOp
+            from qiskit.algorithms import NumPyEigensolver
+
+            qubit_op = PauliSumOp.from_list(
+                [
+                    ("II", -1.052373245772859),
+                    ("ZI", 0.39793742484318045),
+                    ("IZ", -0.39793742484318045),
+                    ("ZZ", -0.01128010425623538),
+                    ("XX", 0.18093119978423156),
+                ]
+            )
+            aux_op1 = PauliSumOp.from_list([("II", 2.0)])
+            aux_op2 = PauliSumOp.from_list([("II", 0.5), ("ZZ", 0.5), ("YY", 0.5), ("XY", -0.5)])
+            aux_ops = {"aux_op1": aux_op1, "aux_op2": aux_op2}
+            pairs = {"names": ["aux_op2"], "indices": [(0, 1), (1, 0), (1, 2)]}
```

Comment:
Would this structure not make more sense?
```suggestion
            pairs = {"aux_op2": [(0, 1), (1, 0), (1, 2)]}
```

Reply:
I thought a little bit about it and my conclusion is that:
- Often the transition amplitudes is only relevant for a predetermined set of indices, for example (0,1), (3,4). This knowledge comes form looking at the adiabatic potential energy surfaces.
- Then for this set of indices, one can be interested in measuring different physical quantities. 

So basically I was trying to find a structure where we don't need to specify the indices for each aux_op because by default, the specified indices would be relevant for all operators. 
However I agree that your suggested structure appears more natural from the user perspective

Reply:
I see. In this case, I would still argue this structure being clearer.
If a user really wants to compute the same index pairs for all ops, he could still do smth like this:
```python
indices = [(0, 1), (3, 4)]
pairs = {"aux_op1": indices, "aux_op2": indices}
```

Reply:
@mrossinek Ok I agree, it is indeed natural to have : Dict[aux_operator] = transition_pairs (of this aux_operator).
However just later in the function, it is best to 1) iterate over all (i,j) and 2) evaluate all operators on one set of (i,j) indices at the time
Would you still advcocate Dict[aux_operator] = transition_pairs?
An alternative is Dict[transition_pair] = aux_operators

Reply:
I'm thinking of just splitting that into two arguments, seems to make more sense while being easy to use